packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -683,6 +695,7 @@ export const Stackbox = () => {
           onSuccess={() => {
             closeModal(ModalId.CONFIRM_STACK);
             openModal(ModalId.SUCCESS_STACK_TOAST);
+            setShouldResetStackbox(true);
```

Comment:
can't we just call a function here, like`resetStackbox()` ?

I find it odd to change the state and have hook that catches that and reset it. Seems like it's more complicated.

Reply:
I think we should call `resetDefaultFormValues()` on those occasions we need to reset the stackbox.
That function shouldn't be tied to an hook, it can be in a hook, but it can also be used here for eg.

For me it makes sense that `onSuccess` I reset the stackbox, so I call `resetDefaultFormValues()` but here we are updating a boolean state that will trigger a `resetDefaultFormValues` somewhere.  

I think this is straight forward behavior, by hiding this complexity on a hook, we're making more difficult to reason on code and be more error prone.

(we should rename `resetDefaultFormValues` to `resetStackbox` or `resetStackboxForm`)

Reply:
Agree with Diogo here, we don't need to listen for a boolean to trigger a reset. Maybe the best solution is to create a Stack values provider, where we provide a `resetStackValues()`.

Reply:
why do we need a provider to do that?

Can't we just have it here `resetStackBox()`, and reset when we need?

Reply:
I also don't think `shouldResetStackbox` makes sense in the strategy provider, it's not related to strategy, it's related to the stackbox.

Reply:
As we spoke offline, I'll create a new context to handle the Stackbox form values separately and consume that context both in the `StrategyCard` (to reset the form values when needed)  and the `Stackbox` component.

I'll resolve this comment thread for now c: